atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1890,8 +1890,23 @@ private void alterGcAndTombstoneInternal(
 
     public void cleanUpSchemaMutationLockTablesState() throws Exception {
         Set<TableReference> tables = lockTables.getAllLockTables();
-        dropTablesInternal(tables);
-        log.info("Dropped all schema mutation lock tables [{}]", tables.toString());
+        clientPool.runWithRetry(client ->
+                batchMutateInternal(client, tables, getMutationsForLockTablesCleanUp(tables), writeConsistency));
+        log.info("Reset the schema mutation locks in tables [{}]", tables.toString());
+    }
+
+    private Map<ByteBuffer, Map<String, List<Mutation>>> getMutationsForLockTablesCleanUp(Set<TableReference> tables) {
```

Comment:
unfortunately, no. Most of the existing code assumes batch mutations are done for a single table whereas we need to use a set of tables here. 😞